web/source/osk/bannerManager.ts

```diff
@@ -18,20 +20,47 @@ namespace com.keyman.osk {
 
     public static readonly DEFAULT_OPTIONS: BannerOptions = {
       persistentBanner: false,
-      enablePredictions: true
+      enablePredictions: true,
+      imagePath: ""
     }
 
     constructor() {
       // Step 1 - establish the container element.  Must come before this.setOptions.
       this.constructContainer();
 
+      // Register a listener for model change events so that we can hot-swap the banner as needed.
+      let keyman = com.keyman.singleton;
+      let device = keyman.util.device;
+
+      // TODO: Remove for production
+      let initialOptions: BannerOptions;
+      if (device.OS == 'iOS') {
+        initialOptions = {
+          persistentBanner: true,
+          enablePredictions: false,
+          imagePath: '../../ios/keyman/Keyman/SWKeyboard/Keyman Banner/banner-Portrait.png'
+        };
+        this.setBanner('image');
+      } else if (device.OS == 'Android') {
+        initialOptions = BannerManager.DEFAULT_OPTIONS;
+        this.setBanner('blank');
+      } else if (device.formFactor == 'desktop') {
+        initialOptions = {
+          persistentBanner: true,
+          enablePredictions: true
+        };
+        this.setBanner('suggestion');
+      } else {
+        initialOptions = BannerManager.DEFAULT_OPTIONS;
+      }
+
       // Initialize with the default options - any 'manually set' options come post-construction.
       // This will also automatically set the default banner in place.
-      this.setOptions(BannerManager.DEFAULT_OPTIONS);
+      this.setOptions(initialOptions);
 
-      // Register a listener for model change events so that we can hot-swap the banner as needed.
-      let keyman = com.keyman.singleton;
       keyman.modelManager['addEventListener']('modelchange', this.selectBanner.bind(this));
+      keyman.modelManager['addEventListener']('invalidatesuggestions', this.invalidateSuggestions.bind(this));
+      keyman.modelManager['addEventListener']('suggestionsready', this.updateSuggestions.bind(this));
```

Comment:
Remove the two added listeners here.  They should only be added for active `SuggestionBanners` as part of the `setBanner` method.

Reply:
Also, please move the `// Register a listener...` comment back above the `'modelchange` event registration.  That's where it belongs.

Reply:
done